lib/MusicBrainz/Server/WebService/JSONSerializer.pm

```diff
@@ -14,53 +15,33 @@ sub fmt { 'json' }
 
 sub serialize
 {
-    my ($self, $type, @data) = @_;
+    my ($self, $type, $entity, $inc, $opts) = @_;
 
     $type =~ tr/-/_/;
 
-    my $override = $self->meta->find_method_by_name($type);
-    return $override->execute($self, @data) if $override;
+    if ($type =~ m/^(.*)_list$/ && exists($entity->{items})) {
+      if (exists($ENTITIES{$1}) && exists($ENTITIES{$1}->{plural_url})) {
+        $type = $1;
+        my $singular = $ENTITIES{$type}->{url};
+        my $plural   = $ENTITIES{$type}->{plural_url};
+        my $ret = {
+          $plural => [ map { serialize_entity($_, $inc, $opts, 1) } sort_by { $_->gid } @{ $entity->{items} } ],
+        };
+        if (defined($entity->{offset}) || defined($entity->{total})) {
+          $ret->{$singular . '-offset'} = number($entity->{offset});
+          $ret->{$singular . '-count' } = number($entity->{total});
+        }
+        return encode_json($ret);
+      }
+    }
 
-    my ($entity, $inc, $opts) = @data;
+    my $override = $self->meta->find_method_by_name($type);
+    return $override->execute($self, $entity, $inc, $opts) if $override;
```

Comment:
Well, is this actually intended for actual overriding, or was it only added to handle lists? At the moment, its only use (as far as I could see) was to handle the lists separately, making it obsolete.
I'm not sure I see the use case otherwise; the default behaviour is to use the specific serialization modules for each entity, which seem to be used by the JSON serializer only. So any overrides of behaviour could just be made there (or, as done now for lists, in the serialize method).
The "just add a method with a particular name to override behaviour" idea is fairly icky, as any existing method could be invoked by accident if a new type string comes in to serialize().